src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,200 @@
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.internal;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import org.junit.jupiter.api.Test;
+import org.xml.sax.Attributes;
+import org.xml.sax.helpers.DefaultHandler;
+
+public class XdocsUrlTest {
+
+    public static final String PACKAGE_NAME =
+            "src/main/java/com/puppycrawl/tools/checkstyle/checks";
+
+    public static final String NODE_NAME = "a";
+
+    public static final String SUFFIX_JAVA = ".java";
+
+    public static final String SUFFIX_CHECK_JAVA = "Check.java";
+
+    public static final String MISC = "misc";
+
+    public static final String ANNOTATION = "annotation";
+
+    public static final String COMMENTS_INDENTATION = "CommentsIndentation";
+
+    public static final String INDENTATION = "Indentation";
+
+    public static final String SUPPRESS_WARNINGS_HOLDER = "SuppressWarningsHolder";
+
+    private static String getRootPath() {
+        return System.getProperty("user.dir");
+    }
+
+    private static String getXmlPath() {
+        return "src/xdocs/checks.xml";
+    }
+
+    private static Map<String, String[]> getXdocsMap() {
+        final String rootPath = getRootPath();
+        final File checks = new File(rootPath + "/" + PACKAGE_NAME);
+        final String[] checksSubDirNames = checks.list((rootDir, pathname) -> {
+            return !pathname.endsWith(SUFFIX_JAVA);
+        });
+        final Map<String, String[]> checksNamesMap = new HashMap<>();
+        if (checksSubDirNames != null) {
+            for (String dirName : checksSubDirNames) {
+                final File dirFile = new File(rootPath + "/" + PACKAGE_NAME + "/" + dirName);
+                final String[] fileNames = dirFile.list((rootDir, pathname) -> {
+                    return pathname.endsWith(SUFFIX_CHECK_JAVA);
+                });
+                checksNamesMap.put(dirName, fileNames);
+            }
+            final String[] miscNames = checks.list((rootDir, pathname) -> {
+                return pathname.endsWith(SUFFIX_CHECK_JAVA);
+            });
+            checksNamesMap.put(MISC, miscNames);
+        }
+        return checksNamesMap;
+    }
+
+    @Test
+    public void testXdocsUrl() throws Exception {
+        final SAXParserFactory parserFactory = SAXParserFactory.newInstance();
+        final SAXParser parser = parserFactory.newSAXParser();
+        final CheckTest checkHandler = new CheckTest(NODE_NAME);
+        try (InputStream input = Files.newInputStream(
```

Comment:
now I use try but not catch block and it is ok. But when I use try-catch, will have a error as follows:
[INFO] --- maven-checkstyle-plugin:3.1.1:check (sevntu-checkstyle-check) @ checkstyle ---
[INFO] There is 1 error reported by Checkstyle 8.29 with /Users/hgh/Desktop/checkstyle/config/checkstyle_sevntu_checks.xml ruleset.
[ERROR] src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java:[99,9] (extension) RequireFailForTryCatchInJunit: try/catch either requires a fail at the end of the try clause for junit tests or to remove the try/catch completely.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 02:03 min
[INFO] Finished at: 2020-04-24T16:16:58+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1:check (sevntu-checkstyle-check) on project checkstyle: You have 1 Checkstyle violation. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.